### PR TITLE
ENH intergrate Jupyter book parts with markdown table generator

### PR DESCRIPTION
**Pull request by htwangtw**
As title, add all the integrations mentioned in PR #24 
- separate neuroview and non-neuroview contributors and acknowledgement pages
- different `make` commands for neuroview and non-neuroview pages

For now, the two sets of files has no difference.
Once the neuroview paper is complete we can either remove the `osf fetch` command or do something else.

### CODE REVIEW

**Review by katjaq — APPROVED** [review]
Heyy! this looks fantastic! and it is mindblowing how fast you are @htwangtw @shawnrhoads and team! 
I made little changes to the text leaving an explanation in the commit message – I am sorry, I think I modified the file that results from the build 🙈🙃 but then I realised and modified the source file as well 🙈🤓 (I changed the introduction to acknowledgement text to not contain specific highlights). So maybe you wanna build again 🙃🙈 sorry! I will learn one day, promise! 🙃

I also changed the makefile – please have a look at it. Line 15 I am pretty positive, line 19 I wasn't sure whether the `neuroview_` needs to be kept (as this contributor list part should literally be the same at the moment and you may have decided to use the same source?).

Thanks so much all! 🚀  ✨

File: Makefile [hunk 1]
```diff
@@ -12,15 +12,15 @@ brainhack_book/neuroview_contributors.md : data/neuroview_contributors_descripti
 	osf -p ${osfid} fetch ${osf_contributors} data/neuroview_contributors.tsv; \
 	python brainhack_book/mdtable.py neuroview contributors; \
 
-brainhack_book/acknowledgments.md : data/acknowledgments.csv data/neuroview_acknowledgements_descriptions.md brainhack_book/mdtable.py
+brainhack_book/acknowledgments.md : data/acknowledgments.csv data/acknowledgements_descriptions.md brainhack_book/mdtable.py
 	python brainhack_book/mdtable.py acknowledgments; \
 
 brainhack_book/contributors.md : data/contributors_descriptions.md  brainhack_book/mdtable.py
-	osf -p ${osfid} fetch ${osf_contributors} data/neuroview_acknowledgments.tsv; \
+	osf -p ${osfid} fetch ${osf_contributors} data/neuroview_contributors.tsv; \
```
Comment (htwangtw) [inline Makefile:19]: Yes this is fine. I thought I changed it....

**Comment by htwangtw** [comment]
It's winter and lockdown so I need something to do on Saturday haha
I think there's a lot we can improve with how I made the script work. Editing the built target can easily be done. Don't worry about it we will find a way to resolve it :+1: 
I checked isuuse #18 and you are right! Happy to remove the duplicates in places and build it locally
BTW the Makefile is only checking dependency and the script now is hardcoded to read relevant files :see_no_evil: Someone knows `Makefile` feel free to jump in and help with this.

**Comment by htwangtw** [comment]
I am getting confused with myself about what pages should be compile and all that, so I am going to update things based on what is most relevant to preprint. Pages referenced in TOC all have their files. We can worry about how exactly we want the contributors and acknowledgement to be handled between the book and the paper.

**Comment by htwangtw** [comment]
Note to myself:

- [ ] change `neuroview_*` to `preprint_*`
- [ ] check `data`

**Comment by htwangtw** [comment]
I broke lot of things while handling all the UK vs US spelling of acknowledg(e)ment :see_no_evil: 
This PR now integrates lots of discussions in PR#26. 

Also to prevent future confusion, I have removed all manually compiled pages and files downloaded from osf. 
- The original spreadsheets are stored/downloaded to `data`.
- The description paragraphs are stored in
  `brainhack_book/*contributors_description.md` and
  `brainhack_book/*acknowledgments_description.md`.
  Any header text changes to the relevant page should be edited there.
- GitHub action will compile the actual markdown page for the website.
  The actual page generated by the script will be named as
  `brainhack_book/*contributors.md` and
  `brainhack_book/*acknowledgments.md`.
- In `brainhack_book/_toc`, the refereced file are the _compiled_ markdown file.
If @katjaq and @shawnrhoads can check the descriptions are all uptodate that will be fantastic!

Also all text related to _neuroview_ has been changed to _preprint_.
I think that's all, ready to go :rocket:

**Review by katjaq — APPROVED** [review]
As always: I am sooo impressed by all the structure you bring to all the different needs, files, formats, and evolving items @htwangtw and @shawnrhoads ! 🚀 ✨ 😍

I made one little change in thank you part of the jupyter book from "manuscript" to "book". Please check if you agree :) Otherwise of course feel free to write manuscript.  
Or it could also be [the benefits described in this manuscript] to "book and manuscript"? You decide :D

Otherwise all green lights to go merge!! 💚💚💚😃

For the contributors' page, I would suggest a change to the `contributors_description.md` : We thank all the individuals who contributed to the creation of the manuscript _and jupyter book_ and the support of Brainhack community towards this project.  
If you agree, please feel free to add that in, or ping me to help.

I am really just sitting here in awe of all your works! 😍🥰💛❤️

Approved!! :D 💚

**Review by PeerHerholz — APPROVED** [review]
Damn, @htwangtw that is some great work here. Thx so much for your dedicated work on these things!

**Review by shawnrhoads — APPROVED** [review]
Truly phenomenal, @htwangtw! 🤩 We are so grateful for all of your hard work and long hours spent making this possible!

I made one minor change for acknowledgment naming consistency (hopefully will help avoid any potential headaches in the future haha).

Only other thought I had: adding the `*_descriptions.md` files from `data/` to `brainhack_book/` will actually build Jupyter Book pages for them (for example: https://brainhack.org/brainhack_jupyter_book/contributors_descriptions.html), but they just wouldn't appear in the TOC. Do we want that?

**Comment by shawnrhoads** [comment]
Oh, also now seeing @Remi-Gau's [great idea](https://mattermost.brainhack.org/brainhack/pl/skstoz6bgb8wpnpkhjw15usmnh) to just remove the "email address" column entirely in the tsv file on OSF. 

Should we remove the lines in `mdtable.py` that remove that particular column?

**Comment by htwangtw** [comment]
>Only other thought I had: adding the *_descriptions.md files from data/ to brainhack_book/ will actually build Jupyter Book pages for them (for example: https://brainhack.org/brainhack_jupyter_book/contributors_descriptions.html), but they just wouldn't appear in the TOC. Do we want that?

Yes thanks for pointing this out! That's a really good point and I didn't think about that - we can add exculsion pattern in the configuration:
https://jupyterbook.org/content/execute.html#exclude-files-from-execution
I will try it out.

>Should we remove the lines in mdtable.py that remove that particular column?
Update: Or add an if-statement to catch whenever that update happens?

That's possible since the column check is already an isolated function-_ish_

**Comment by htwangtw** [comment]
## summary of the final changes
- `mdtable.py` now has a CLI. Any file input will be done with the `Makefile` (so all typo issue would be file name and/or file name referenced in `Makefile`)
- exclude `*descriptions.md` from book building workflow
- remove files generated by `make` and files host on osf in the repo

Any enhancement should be handled with new PR

**Pull request merged**